lib/kafka_rest_client.js

```diff
@@ -272,9 +272,9 @@ KafkaRestClient.prototype.tryMakeRequest = function tryMakeRequest(produceMessag
             self.statsd.timing(metricPrefix + '.latency', time);
         }
 
-        if (err) {
+        if (err || (res && res.statusCode >= 500 && res.statusCode < 600)) {
```

Comment:
Two questions: 
1. will 'res' be null? if so, I think we also need to check it in the 'else if' branch below.
2. is it possible 'err' is not null but res.statusCode is say 404. Then this if-branch is hit and will do retry. But I guess we should drop msg for 404 for non-hp topics.